src/mem/slicc/symbols/StateMachine.py

```diff
@@ -302,38 +304,75 @@ class $py_ident(RubyController):
                 )
 
         code.dedent()
-        code.write(path, f"{py_ident}.py")
+
+        # Needed for backwards compatibility while there is only one protocol
+        # compiled. When moving to multiple protocols in the gem5 binary, this
+        # will need to change.
```

Comment:
Hmm, is this comment still correct?  Does it imply that currently the PR doesn't currently support multiple protocols?

Reply:
I'll update this comment. The code is correct, but the comment is out of date.

Reply:
For rebasing reasons, this comment isn't going to be updated in the right commit, but the new text will be right after the merge.